### ISSUE TITLE
Top 25 OpInfos for functorch

### ISSUE THREAD

**Issue by zou3519** [pytorch/functorch#78]
We'd love help on these.

The check box is for if the OpInfo has added to PyTorch core. The ultimate goal is for all of these OpInfos to exist in PyTorch core.
The OpInfo is bolded if we have a poor man's version* of the OpInfo in the functorch repo (see https://github.com/facebookresearch/functorch/blob/main/test/functorch_additional_op_db.py).

Exists
- [x] **torch.nn.functional.softmax** (https://github.com/pytorch/pytorch/pull/62077)
- [x] **torch.nn.functional.relu** (https://github.com/pytorch/pytorch/pull/62076)
- [x] **torch.nn.functional.interpolate** (https://github.com/pytorch/pytorch/pull/61956)
- [x] **torch.nn.functional.pad** (https://github.com/pytorch/pytorch/pull/62814)
- [x] **torch.nn.functional.normalize** (https://github.com/pytorch/pytorch/pull/62635)
- [x] **torch.nn.functional.cross_entropy** (https://github.com/pytorch/pytorch/pull/63547)
- [x] torch.nn.functional.grid_sample (https://github.com/pytorch/pytorch/pull/62311)
- [x] torch.nn.functional.one_hot (https://github.com/pytorch/pytorch/pull/62253)
- [x] torch.nn.functional.mse_loss
- [x] **torch.nn.functional.conv2d** (https://github.com/pytorch/pytorch/pull/63517)
- [x] torch.nn.functional.dropout (https://github.com/pytorch/pytorch/pull/62315)
- [x] torch.nn.functional.softplus (https://github.com/pytorch/pytorch/pull/62317)
- [x]  **torch.nn.functional.linear** (https://github.com/pytorch/pytorch/pull/61971)
- [x] torch.nn.functional.avg_pool2d (https://github.com/pytorch/pytorch/pull/62455)
- [x] torch.nn.functional.max_pool2d (https://github.com/pytorch/pytorch/pull/63530)
- [x] torch.nn.functional.nll_loss (https://github.com/pytorch/pytorch/pull/64203)
- [x] torch.nn.functional.embedding (https://github.com/pytorch/pytorch/pull/63633)
- [x]  torch.nn.functional.adaptive_avg_pool2d (https://github.com/pytorch/pytorch/pull/62704)
- [x] torch.nn.functional.cosine_similarity (https://github.com/pytorch/pytorch/pull/62959)
- [x] torch.nn.functional.unfold https://github.com/pytorch/pytorch/pull/62705
- [x] torch.nn.functional.batch_norm (https://github.com/pytorch/pytorch/pull/63218)
- [x] torch.nn.functional.conv_transpose2d https://github.com/pytorch/pytorch/pull/62882
- [x] torch.nn.functional.layer_norm https://github.com/pytorch/pytorch/pull/63276

*Why do we have poor man's version of these OpInfos? It's because right now we only care about float32 sample inputs on CPU and CUDA and OpInfos have a lot of flags that take some time to tweak.

**Comment by Chillee** [comment]
We are also implementing many of these OpInfos, so you can contribute by looking at these (https://github.com/facebookresearch/functorch/blob/main/test/functorch_additional_op_db.py) and porting them to PyTorch core.

**Comment by pmeier** [comment]
I'll take up `grid_sample`, `mse_loss`, and `one_hot`.

**Comment by krshrimali** [comment]
Planning to take up: `avg_pool2d`, `max_pool2d`, `normalize` today.

**Comment by Chillee** [comment]
@krshrimali @pmeier Thanks for the exhaustive OpInfos - they've already been quite useful :) Concretely, I found a bug in our `softmax/log_softmax` batching rules thanks to the softmax OpInfo (https://github.com/facebookresearch/functorch/commit/13421bc4d965701a46c4f9b374c1ec0f822ec389), and the `mse_loss` OpInfo was very helpful for writing a batching rule (https://github.com/facebookresearch/functorch/commit/02f7908cbf18786f7bd71e731016a8a72783691f)  and having me be confident that I got all the (somewhat tricky) cases right.

**Comment by krshrimali** [comment]
I'm planning to pick: `conv_transpose2d, adaptive_avg_pool2d, batch_norm, layer_norm, cosine_similarity` for the coming 2 days.

For `max_pool2d`, there is a PR open to fix its behavior for `return_indices` (args ordering) - which once merged will allow me to not skip tests in OpInfo for `max_pool2d`. Though, if it takes time, I'll open a PR for OpInfo of `max_pool2d` this week and maybe skip relevant tests for now. Thanks!

**Comment by krshrimali** [comment]
Picking the last set of OpInfos coming week: `cross_entropy, conv2d, max_pool2d, embedding`. `dropout` is blocked on a few issues, should be resolved coming week. Everything else has an open/merged PR if I'm not wrong. :)

**Comment by pmeier** [comment]
I have some local commits for nll_loss. If no one has worked on it, I will next week.

**Comment by kshitij12345** [comment]
Picking up `conv2d`

**Comment by AnirudhDagar** [comment]
I'll take `max_pool2d`
cc @krshrimali

**Comment by zou3519** [comment]
We did it! Thank you @kshitij12345 @pmeier @krshrimali for completing these OpInfos and reminding me to land them.
